scripts/analysis/get-all-components.ts

```diff
@@ -0,0 +1,138 @@
+import { join } from 'path';
+import { readdir, stat, readdirSync, readFileSync } from 'fs';
+import { promisify } from 'util';
+import { packageNames } from '../package-names';
+import { paths } from '../paths';
+import { parse } from '@typescript-eslint/typescript-estree';
+
+const readdirP = promisify(readdir);
+const statP = promisify(stat);
+
+const coreComponents = join(__dirname, '../../src/core/components');
+const editorialComponents = join(
```

Comment:
TODO: refactor to scan `packages/@guardian/source-react-components-development-kitchen`

Reply:
TO-DONE ✅